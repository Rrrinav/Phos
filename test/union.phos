union State {
    let ok: bool;
    let err: string;
}

let res : State = State::ok(true);
let res2 := State::err("Some unexpected error occured");

if (res.has(State::ok)) {
    print("State res is ok");
} else {
    print("State res is no ok: " + res.err);
}

if (res2.has(State::ok)) {
    print("State res2 is ok");
} else {
    print("State res2 is no ok: "+ res2.err);
}

model State_getter {
    let state: State;

    static fn get_ok() -> State {
        return State::ok(true);
    }

    static fn get_err() -> State {
        return State::err("Some err");
    }
}

let ok := State_getter::get_ok();
let err := State_getter::get_err();

print(ok.ok);
print(err.err);
