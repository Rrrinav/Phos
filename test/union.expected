State res is ok
State res2 is no ok: Some unexpected error occured
true
Some err
